pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -214,12 +214,20 @@ public void updateIngestionDelay(long ingestionTimeMs, long firstStreamIngestion
       // Do not update the ingestion delay metrics during server startup period
       return;
     }
+    if ((ingestionTimeMs < 0) && (firstStreamIngestionTimeMs < 0)) {
```

Comment:
should this conditional be an `||` instead of `&&` ?

Reply:
&& is the intention:

Case 1 Stream does not support ingestion delay nor end to end delay: we bail out early and do not record anything.

If either ingestion delay or end to end ingestion delay we do record the times but only publish the one that has value larger than zero.

I think in the previous incarnation the code must have been dumping negative metrics or very large number for any ingestion delay not supported.

Reply:
ok it was not obvious reading just this snippet of code.